R/flywire-skeleton.R

```diff
@@ -0,0 +1,371 @@
+                        sampling_dist=500,
+                        heal=TRUE,
+                        heal.k=10L,
+                        heal.threshold=Inf,
+                        reroot = TRUE,
+                        k.soma.search = 10,
+                        radius.soma.search = 2500,
+                        brain = NULL,
+                        ...){
+  if(is.null(tryCatch(reticulate::py$vol,error=function(e) NULL))){
+    py_skel_imports(cloudvolume.url=cloudvolume.url)
+  }
+  if(grepl("obj$",id)){
+    reticulate::py_run_string(sprintf("m=tm.load_mesh('%s',process=False)",id), ...)
+    if(mesh3d){
+      mesh = readobj::read.obj(id)
+    }
+  }else{
+    reticulate::py_run_string(sprintf("id=%s",id), ...)
+    reticulate::py_run_string("m = vol.mesh.get(id, deduplicate_chunk_boundaries=False)[id]", ...)
+    mesh = NULL
+  }
+  reticulate::py_run_string("m = tm.Trimesh(m.vertices, m.faces)", ...)
+  reticulate::py_run_string(sprintf("simp = sk.simplify(m, ratio=%s)",ratio), ...)
+  reticulate::py_run_string(sprintf("cntr = sk.contract(simp, SL=%s, WH0=%s, iter_lim=%s, epsilon=%s, precision=%s, validate=%s, progress=False)",
+                                    SL,WHO,iter_lim,epsilon,precision,ifelse(validate,"True","False")), ...)
+  reticulate::py_run_string(sprintf("swc = sk.skeletonize(cntr, method='%s', sampling_dist=500, progress=False)",
+                                    method, sampling_dist), ...)
+  if(clean){
+    reticulate::py_run_string("swc = sk.clean(swc, simp)", ...)
+  }
+  if(radius){
+   reticulate::py_run_string(sprintf("swc['radius'] = sk.radii(swc, simp, method='%s', n=5, aggregate='mean')",method.radii), ...)
+  }else{
+    reticulate::py_run_string("swc['radius'] = 0", ...)
+  }
+  reticulate::py_run_string("for c in ['x', 'y', 'z']: swc[c] = swc[c].astype(int)", ...)
+  swc = reticulate::py$swc
+  colnames(swc) = c("PointNo","Parent","X","Y","Z","W")
+  neuron = nat::as.neuron(swc)
+  if(heal){
+    neuron = suppressMessages(nat::stitch_neurons_mst(x = neuron, threshold = heal.threshold, k = heal.k))
+  }
+  if(reroot){
+    neuron = reroot_hairball(neuron, k.soma.search = k.soma.search, radius.soma.search = radius.soma.search)
+  }
+  if(mesh3d){
+    if(is.null(mesh)){
+      savedir <- tempdir()
+      ff=file.path(savedir, paste0(id, '.obj'))
+      reticulate::py_run_string(sprintf("m.export('%s')",ff), ...)
+      res=sapply(ff, nat::read.neurons)
+      mesh=res[[1]][[1]]
+      neuron$mesh3d = mesh
+      class(neuron) = c(class(neuron), "neuronmesh")
+    }
+  }
+  neuron
+}
+
+
+#' @rdname skeletor
+#' @export
+reroot_hairball <- function(x,
+                           k.soma.search = 10,
+                           radius.soma.search = 2500,
+                           brain = NULL){
+
+  # Get end and branch points, as vectors
+  e = nat::endpoints(x)
+  if(!is.null(brain)){
+    pin = !nat::pointsinside(x = nat::xyzmatrix(x$d), surf = brain)
```

Comment:
pointsinside automatically does xyzmatrix on x. The only reason it might not work would be if you had assigned `neuronmesh` class as the sole class for your neurons (they should also have class `neuron` as a base class)